Mathlib/Data/Polynomial/AlgebraMap.lean

```diff
@@ -91,10 +91,11 @@ theorem algHom_ext' [Algebra R A'] [Algebra R B'] {f g : A'[X] →ₐ[R] B'}
 
 variable (R)
 
-/-- Algebra isomorphism between `R[X]` and `AddMonoidAlgebra R ℕ`. This is just an
+open AddMonoidAlgebra in
+/-- Algebra isomorphism between `R[X]` and `R[ℕ]`. This is just an
 implementation detail, but it can be useful to transfer results from `Finsupp` to polynomials. -/
 @[simps!]
-def toFinsuppIsoAlg : R[X] ≃ₐ[R] AddMonoidAlgebra R ℕ :=
+def toFinsuppIsoAlg : R[X] ≃ₐ[R] R[ℕ] :=
```

Comment:
Interesting that this works and Lean doesn't try to find `(X : Type*)`.

I think it's probably safe to use the same notation for `AddMonoidAlgebra` and `MonoidAlgebra`, as long as they're scoped in different locales(=namespaces now?), since it's not common that we open both (and other conflicts will appear if we do). I imagine if we do any Kaplansky's conjecture stuff in mathlib we'd be using `MonoidAlgebra` rather than `AddMonoidAlgebra` (we have to make a choice to avoid duplication unless we make to_additive work), although [existing works](https://leanprover.zulipchat.com/#narrow/stream/270676-lean4/topic/Formalizing.20Gardam's.20disproof.20of.20a.20.22Kaplansky.20Conjecture.22) seem to use hand-rolled computable versions of group algebra.